extensions/wsc/src/main/java/org/mule/extension/ws/internal/WscExceptionEnricher.java

```diff
@@ -30,9 +32,19 @@
   public Exception enrichException(Exception e) {
     if (e instanceof SoapFault) {
       SoapFault sf = (SoapFault) e;
+      // TODO build proper error type.
       return new SoapFaultException(sf.getFaultCode(), sf.getSubCode(), sf.getMessage(), sf.getDetail());
     }
     if (e instanceof WscException) {
+      // TODO build proper error type.
+      return e;
+    }
+    if (e instanceof WscEncodingException) {
+      // TODO build proper error type.
+      return e;
+    }
+    if (e instanceof InvalidWsdlException) {
+      // TODO build proper error type.
```

Comment:
this will fall into an uber set of tickets we'll do to use steve's magic in all the extensions.